be/src/runtime/sorted_chunks_merger.cpp

```diff
@@ -347,4 +347,51 @@ Status CascadeChunkMerger::get_next(ChunkUniquePtr* output, std::atomic<bool>* e
     return Status::OK();
 }
 
+ConstChunkMerger::ConstChunkMerger(RuntimeState* state) : ChunkMerger(state) {}
+
+Status ConstChunkMerger::init(const std::vector<ChunkProvider>& providers, const std::vector<ExprContext*>* sort_exprs,
+                              const SortDescs& sort_desc) {
+    for (const auto& expr : *sort_exprs) {
+        DCHECK(expr->root()->is_constant());
+    }
+    _providers = providers;
+    return Status::OK();
+}
+Status ConstChunkMerger::init(const std::vector<ChunkProvider>& providers, const std::vector<ExprContext*>* sort_exprs,
+                              const std::vector<bool>* sort_orders, const std::vector<bool>* null_firsts) {
+    for (const auto& expr : *sort_exprs) {
+        DCHECK(expr->root()->is_constant());
+    }
+    _providers = providers;
+    return Status::OK();
+}
+
+bool ConstChunkMerger::is_data_ready() {
```

Comment:
const

Reply:
Just keep same interface, I'm not sure other impelemention is const